src/author/author.service.ts

```diff
@@ -31,32 +34,35 @@ export class AuthorService {
     const { name } = updateDto;
     const found = await this.getAuthorById(id);
     if (found) {
-      console.log(found);
-      return await this.authorRepository.save({
-        id: id,
-        name: name,
-      });
+      found.name = name;
+      return await this.authorRepository.save(found);
     }
   }
   async deleteAuthor(id: string): Promise<string> {
     const found = await this.getAuthorById(id);
     if (!found) {
       throw new NotFoundException(`${Message.NOT_FOUND.author}=${id}`);
     } else {
-      await this.authorRepository.delete({ id });
-      return 'delete succes ! ';
+      found.isDelete = true;
+      try {
+        await this.authorRepository.save(found);
+        return 'delete succes ! ';
+      } catch (error) {
+        console.log(error);
+      }
     }
   }
   async createAuthor(name: string): Promise<Author> {
     const newAuthor = {
       name: name,
+      isDelete: false,
```

Comment:
Set default value of isDeleted is false in the Entity.

Reply:
done